dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/index.ts

```diff
@@ -1,36 +1,61 @@
 import { HttpStatusCode } from 'axios';
 import { Response } from 'express';
+import { FindOptionsWhere, Like } from 'typeorm';
+import { UtilisationReportEntity } from '@ukef/dtfs2-common';
 import { UtilisationReportRepo } from '../../../../repositories/utilisation-reports-repo';
 import { UtilisationReportReconciliationDetails } from '../../../../types/utilisation-reports';
 import { CustomExpressRequest } from '../../../../types/custom-express-request';
 import { NotFoundError, ApiError } from '../../../../errors';
 import { mapUtilisationReportEntityToReconciliationDetails } from './helpers';
+import { REGEX } from '../../../../constants';
 
 export type GetUtilisationReportReconciliationDetailsByIdRequest = CustomExpressRequest<{
   params: {
     reportId: string;
   };
+  query: {
+    facilityIdQuery?: string;
+  };
 }>;
 
 type ResponseBody = UtilisationReportReconciliationDetails | string;
 
 export const getUtilisationReportReconciliationDetailsById = async (req: GetUtilisationReportReconciliationDetailsByIdRequest, res: Response<ResponseBody>) => {
   const { reportId } = req.params;
+  const { facilityIdQuery } = req.query;
 
   try {
+    const feeRecordFindOptions: FindOptionsWhere<UtilisationReportEntity> =
+      facilityIdQuery && REGEX.UKEF_PARTIAL_FACILITY_ID_REGEX.test(facilityIdQuery)
+        ? {
+            feeRecords: {
+              facilityId: Like(`%${facilityIdQuery}%`),
+            },
+          }
+        : {};
+
     const utilisationReport = await UtilisationReportRepo.findOne({
-      where: { id: Number(reportId) },
       relations: { feeRecords: true },
+      where: {
+        id: Number(reportId),
+        ...feeRecordFindOptions,
+      },
     });
+
     if (!utilisationReport) {
-      throw new NotFoundError(`Failed to find a report with id '${reportId}'`);
+      const errorMessage = facilityIdQuery
+        ? `Failed to find a report with id '${reportId}' & facility id containing '${facilityIdQuery}'`
+        : `Failed to find a report with id '${reportId}'`;
+      throw new NotFoundError(errorMessage);
```

Comment:
What happens if the report exists but has no fee records matching the facility id search term?

I was expecting something along the lines of the page to still load but the table of fee records to be replaced with a message about there not being any matching the search term, but I can't see that in the template, although I do appreciate there is not an actual design for this attached to the ticket.